src/rtctools_heat_network/pycml/component_library/heat/gas/gas_pipe.py

```diff
@@ -34,15 +32,15 @@ def __init__(self, name, **modifiers):
         self.add_variable(Variable, "dH", nominal=self.Q_nominal * self.r)
 
         # head is lost over the pipe
-        self.add_equation(((self.GasOut.H - (self.GasIn.H - self.dH)) / (self.Q_nominal*self.r)))
+        self.add_equation(((self.GasOut.H - (self.GasIn.H - self.dH)) / (self.Q_nominal * self.r)))
         # for now simple linear head loss
-        self.add_equation(((self.dH - self.GasIn.Q * self.r) / (self.Q_nominal*self.r)))
+        self.add_equation(((self.dH - self.GasIn.Q * self.r) / (self.Q_nominal * self.r)))
         # Flow should be preserved
         self.add_equation(((self.GasIn.Q - self.GasOut.Q) / self.Q_nominal))
-        self.add_equation(((self.GasIn.Q) / self.Q_nominal))
+        self.add_equation(((self.GasIn.Q - self.GasIn.mass_flow / self.density) / self.Q_nominal))
```

Comment:
Added this line such that there is a link between the volumetric flow and mass flow to ensure that also a pressure drop is calculated,  otherwise another gas test fails. In future this needs to be updated.

Reply:
@jimrojerTNO did you maybe check this impact yet?

Reply:
This in principle is a good thing to do. However, for the results we created for Javier we left this constraint out so that the gas network would not be limiting.

Reply:
@jimrojerTNO we will then keep the link in the code for volumetric flow and mass flow?

Reply:
I think that is our best option indeed, as that is physically better.